[c89/IIO] Getting compile errors when c89 is set instead of gnu89

When c89 is set instead of gnu89 for the 'c_std' option (https://github.com/nnsuite/nnstreamer/pull/1213/files), we could get some compile errors as follows:

```bash
[18:06:24-wook]~/Work/prj_nnstreamer/nnstreamer/build  (prepare-c89) $ ninja
[2/20] Compiling C object 'gst/nnstreamer/nnstreamer@sta/tensor_src_iio.c.o'.
FAILED: gst/nnstreamer/nnstreamer@sta/tensor_src_iio.c.o 
ccache cc  -Igst/nnstreamer/nnstreamer@sta -Igst/nnstreamer -I../gst/nnstreamer -Igst/nnstreamer/tensor_transform -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/gstreamer-1.0 -I/usr/lib/x86_64-linux-gnu/gstreamer-1.0/include -I/usr/include/orc-0.4 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=c89 -O0 -g '-DVERSION="0.1.3"' -DHAVE_ORC=1 -DENABLE_TENSORFLOW=1 -DENABLE_TENSORFLOW_LITE=1 -fPIC -pthread -MMD -MQ 'gst/nnstreamer/nnstreamer@sta/tensor_src_iio.c.o' -MF 'gst/nnstreamer/nnstreamer@sta/tensor_src_iio.c.o.d' -o 'gst/nnstreamer/nnstreamer@sta/tensor_src_iio.c.o' -c /home/wook/Work/prj_nnstreamer/nnstreamer/gst/nnstreamer/tensor_source/tensor_src_iio.c
/home/wook/Work/prj_nnstreamer/nnstreamer/gst/nnstreamer/tensor_source/tensor_src_iio.c: In function ‘gst_tensor_src_iio_start’:
/home/wook/Work/prj_nnstreamer/nnstreamer/gst/nnstreamer/tensor_source/tensor_src_iio.c:1464:30: error: implicit declaration of function ‘fileno’ [-Werror=implicit-function-declaration]
   self->buffer_data_fp->fd = fileno (self->buffer_data_file);
                              ^
/home/wook/Work/prj_nnstreamer/nnstreamer/gst/nnstreamer/tensor_source/tensor_src_iio.c: In function ‘gst_tensor_src_iio_process_scanned_data’:
/home/wook/Work/prj_nnstreamer/nnstreamer/gst/nnstreamer/tensor_source/tensor_src_iio.c:1755:19: error: implicit declaration of function ‘be16toh’ [-Werror=implicit-function-declaration]
           value = be16toh (value);
                   ^
/home/wook/Work/prj_nnstreamer/nnstreamer/gst/nnstreamer/tensor_source/tensor_src_iio.c:1759:19: error: implicit declaration of function ‘le16toh’ [-Werror=implicit-function-declaration]
           value = le16toh (value);
                   ^
/home/wook/Work/prj_nnstreamer/nnstreamer/gst/nnstreamer/tensor_source/tensor_src_iio.c:1774:19: error: implicit declaration of function ‘be32toh’ [-Werror=implicit-function-declaration]
           value = be32toh (value);
                   ^
/home/wook/Work/prj_nnstreamer/nnstreamer/gst/nnstreamer/tensor_source/tensor_src_iio.c:1778:19: error: implicit declaration of function ‘le32toh’ [-Werror=implicit-function-declaration]
           value = le32toh (value);
                   ^
/home/wook/Work/prj_nnstreamer/nnstreamer/gst/nnstreamer/tensor_source/tensor_src_iio.c:1797:19: error: implicit declaration of function ‘be64toh’ [-Werror=implicit-function-declaration]
           value = be64toh (value);
                   ^
/home/wook/Work/prj_nnstreamer/nnstreamer/gst/nnstreamer/tensor_source/tensor_src_iio.c:1801:19: error: implicit declaration of function ‘le64toh’ [-Werror=implicit-function-declaration]
           value = le64toh (value);
                   ^
cc1: all warnings being treated as errors
[3/20] Compiling C object 'gst/nnstreamer/nnstreamer@sha/tensor_src_iio.c.o'.
FAILED: gst/nnstreamer/nnstreamer@sha/tensor_src_iio.c.o 
```

By adding the below code snippet , those errors can be easily fixed but I am not sure that this way is right.

In short, 
- [x] 1. the functions used for byte order conversion (be16toh, le16toh, be32toh, le32toh, be64toh and le64toh) [are nonstandard](http://man7.org/linux/man-pages/man3/htole32.3.html).
  - Fixed: https://github.com/nnsuite/nnstreamer/pull/1294

- [x] 2. fileno requires [_POSIX_C_SOURCE >= 1 || _XOPEN_SOURCE || _POSIX_SOURCE](https://linux.die.net/man/3/fileno)
  - Fixed: https://github.com/nnsuite/nnstreamer/pull/1302

See also: https://github.com/nnsuite/nnstreamer/pull/1213 
Related issue: https://github.com/nnsuite/nnstreamer/issues/1199

--- a/gst/nnstreamer/tensor_source/tensor_src_iio.c
+++ b/gst/nnstreamer/tensor_source/tensor_src_iio.c
@@ -1748,11 +1748,11 @@ gst_tensor_src_iio_process_scanned_data (GList * channels, gchar * data,
       {
         guint16 value = *(guint16 *) (data + prop->location);
         if (prop->big_endian) {
-          value = be16toh (value);
+          value = GUINT16_FROM_BE (value);
           /** right shift the extra storage bits for big endian */
           value >>= (16 - prop->storage_bits);
         } else {
-          value = le16toh (value);
+          value = GUINT16_FROM_LE (value);
           /** mask out the extra storage bits for little endian */
           storage_mask = (G_GUINT64_CONSTANT (1) << prop->storage_bits) - 1;
           value &= storage_mask;
@@ -1767,11 +1767,11 @@ gst_tensor_src_iio_process_scanned_data (GList * channels, gchar * data,
       {
         guint32 value = *(guint32 *) (data + prop->location);
         if (prop->big_endian) {
-          value = be32toh (value);
+          value = GUINT32_FROM_BE (value);
           /** right shift the extra storage bits for big endian */
           value >>= (32 - prop->storage_bits);
         } else {
-          value = le32toh (value);
+          value = GUINT32_FROM_LE (value);
           /** mask out the extra storage bits for little endian */
           storage_mask = (G_GUINT64_CONSTANT (1) << prop->storage_bits) - 1;
           value &= storage_mask;
@@ -1790,11 +1790,11 @@ gst_tensor_src_iio_process_scanned_data (GList * channels, gchar * data,
       {
         guint64 value = *(guint64 *) (data + prop->location);
         if (prop->big_endian) {
-          value = be64toh (value);
+          value = GUINT64_FROM_BE (value);
           /** right shift the extra storage bits for big endian */
           value >>= (64 - prop->storage_bits);
         } else {
-          value = le64toh (value);
+          value = GUINT64_FROM_LE (value);
           /** mask out the extra storage bits for little endian */
           storage_mask = (G_GUINT64_CONSTANT (1) << prop->storage_bits) - 1;
           value &= storage_mask;

